### PR TITLE
[full-ci] change e2e test for enforce pass

### PR DESCRIPTION
**Pull request by ScharfViktor**
- [x] enable enforced public link password for e2e tests
- [x] I disabled enforced public link password for old UI-tests.
- [x] fixed flaky https://github.com/owncloud/web/issues/9941
- [x] added checking that user is in the file viewer https://github.com/owncloud/web/pull/9821#discussion_r1377057491   
- [x] some small improvements

ToDo: get rid of:
- all warnings related `pnpm lint`
- page.locator.isVisible() - flaky https://playwright.dev/docs/api/class-locator#locator-is-visible 
- deprecated methods like `page.waitForNavigation()`

### CODE REVIEW

**Review by JammingBen — APPROVED** [review]

File: .drone.star [hunk 1]
```diff
@@ -1263,7 +1263,7 @@ def webService():
         ],
     }]
 
-def ocisService(type, tika_enabled = False):
+def ocisService(type, tika_enabled = False, enforce_password_public_link_disabled = False):
```
Comment (saw-jan) [inline .drone.star:1266]:
```suggestion
def ocisService(type, tika_enabled = False, enforce_password_public_link = False):
```

File: tests/e2e/support/objects/app-files/utils/editor.ts [hunk 2]
```diff
@@ -16,3 +19,22 @@ export const save = async (page: Page): Promise<unknown> => {
     page.locator(saveTextEditorOrViewerButton).click()
   ])
 }
+
+export const isOpen = async ({
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/utils/editor.ts:23]:
```suggestion
export const fileViewerLocator = async ({
```

File: tests/e2e/support/objects/app-files/utils/editor.ts [hunk 3]
```diff
@@ -16,3 +19,22 @@ export const save = async (page: Page): Promise<unknown> => {
...
+  fileViewerType: string
+}): Promise<Locator> => {
+  switch (fileViewerType) {
+    case 'text-editor':
+      return await page.locator(texEditor)
+    case 'pdf-viewer':
+      return await page.locator(pdfViewer)
+    case 'image-viewer':
+      return await page.locator(imageViewer)
+    default:
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/utils/editor.ts:37]:
```suggestion
    case 'text-editor':
      return page.locator(texEditor)
    case 'pdf-viewer':
      return page.locator(pdfViewer)
    case 'image-viewer':
      return page.locator(imageViewer)
    default:
```

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Pull request merged**